app/uk/gov/hmrc/serviceconfigs/persistence/DeploymentConfigSnapshotRepository.scala

```diff
@@ -66,16 +80,136 @@ class DeploymentConfigSnapshotRepository @Inject()(
       .toFuture
       .map(_ => ())
 
+  def removeLatestFlagForNonDeletedSnapshotsInEnvironment(environment: Environment): Future[Unit] =
+    collection
+      .updateMany(
+        filter = and(
+          equal("deploymentConfig.environment", environment.asString),
+          equal("latest", true),
+          equal("deleted", false),
+        ),
+        update = set("latest", false)
+      ).toFuture().map(_ => ())
+
+  def removeLatestFlagForServiceInEnvironment(serviceName: String, environment: Environment): Future[Unit] =
+    collection
+      .updateMany(
+        filter = and(
+          equal("deploymentConfig.name", serviceName),
+          equal("deploymentConfig.environment", environment.asString),
+        ),
+        update = set("latest", false)
+      ).toFuture().map(_ => ())
+
+
   def populate(date: Instant): Future[Unit] = {
 
     def forEnvironment(environment: Environment): Future[Unit] = {
       for {
         deploymentConfigs <- deploymentConfigRepository.findAll(environment)
-        snapshots         =  deploymentConfigs.map(DeploymentConfigSnapshot(date, _))
-        _                 <- collection.insertMany(snapshots).toFuture()
+        latestSnapshots   <- latestSnapshotsInEnvironment(environment)
+        planOfWork        =  PlanOfWork.fromLatestSnapshotsAndCurrentDeploymentConfigs(latestSnapshots.toList, deploymentConfigs.toList, date)
+        _                 <- executePlanOfWork(planOfWork, environment)
       } yield ()
     }
 
     Environment.values.foldLeftM(())((_, env) => forEnvironment(env))
   }
+
+  def executePlanOfWork(planOfWork: PlanOfWork, environment: Environment): Future[Unit] = {
+    logger.debug(s"Processing `DeploymentConfigSnapshot`s for ${environment.asString}")
+    val batchInsertions =
+      planOfWork.snapshots.map(_.deploymentConfigSnapshot) ++
+      planOfWork.snapshotSynthesisedDeletions.map(_.deploymentConfigSnapshot)
+
+    def reintroduceServiceSnapshot(snapshotServiceReintroduction: SnapshotServiceReintroduction) = {
```

Comment:
Since only the `deploymentConfigSnapshot` of `snapshotServiceReintroduction` is used, if the function takes the `DeploymentConfigSnapshot` it keeps the `for-comprehension` operating on the same type for all.
It makes me wonder if `planOfWork` should only take `DeploymentConfigSnapshot` avoiding all the `.map(_.deploymenConfigSnapshot)`
Actually, It looks like the wrappers are just for type safety? Are they pulling their weight? Would seem simpler to just trust the property names of `planOfWork`

Reply:
This has been a really good change. You're right that they weren't pulling their weight. Cheers!